boot/bootstate20.go

```diff
@@ -810,3 +924,67 @@ func (bsmark *bootState20MarkSuccessful) commit() error {
 
 	return nil
 }
+
+// genericInitramfsSelectSnap will run the logic to choose which snap should be
+// mounted during the initramfs using the given bootState and the expected try
+// status. The try status is needed because during the initramfs we will have
+// different statuses for kernel vs base snaps, where base snap is expected to
+// be in "try" mode, but kernel is expected to be in "trying" mode. It returns
+// the first and second choice for what snaps to mount, and the bool indicates
+// if there is a second snap set or not. If there is a second snap, then that
+// snap is the fallback or non-trying snap and the first snap is the try snap.
+func genericInitramfsSelectSnap(bs bootState, expectedTryStatus, typeString string) (
+	firstChoice, secondChoice snap.PlaceInfo,
+	fallbackExpected bool,
```

Comment:
looking at this again, do we need the flag,  it's basically  second != nil (if no error),  I'm unsure whether the flag clarifies or obscures things. for sure here it's easier to not have the flag

Reply:
Yeah, I agree the flag is bit confusing when jumping between this place and where the function gets called in respective `selectAndCommitSnapInitramfsMount()`

Reply:
we have now come full circle on this hooray

:arrow_right: :arrow_right: :arrow_down: 
:arrow_up: :rainbow:  :arrow_down: 
:arrow_up: :arrow_left: :arrow_left: 

sure I can unrefactor that argument and just have the callers do `second != nil`.

Reply:
not really :) , my first proposal which probably wasn't clear, was for a flag as input for  snapTryStatus == expectedTryStatus (not output) but that doesn't work because of where the loading happens which I missed when I made the proposal

Reply:
> I notice that the comment doesn't mention the expectedTryStatus, I also wonder if that should 
> be more something like expectFallback bool and let the caller deal with what it means

I wanted `expectedTryStatus string` to be `expectFallback bool`

Reply:
well in any case I don't have the argument here anymore so hopefully it is to everyone's tastes :-)